src/OpenSage.Game/Logic/Object/GameObject.cs

```diff
@@ -403,6 +405,19 @@ internal void LogicTick(ulong frame, in TimeInterval time)
             }
         }
 
+        public bool CanRecruitHero(ObjectDefinition definition)
+        {
+            foreach (var obj in Parent.Items)
+            {
+                if (obj.Definition.Name == definition.Name
```

Comment:
Why can we not compare object references? I thought we had a unique `ObjectDefinition` instance per name.

Reply:
done